src/Grid/index.jsx

```diff
@@ -36,138 +47,112 @@ export default class Grid extends React.Component
+        rows   : PropTypes.number,
     };
 
     static defaultProps =
     {
-        align         : 'auto',
+        align         : 'top',
+        autoCols      : undefined,
+        autoFlow      : 'row',
+        autoRows      : undefined,
         children      : undefined,
         className     : undefined,
-        gutters       : 'M',
-        hasWrap       : true,
-        noWarn        : false,
-        onClick       : undefined,
-        onMouseOut    : undefined,
-        onMouseOver   : undefined,
+        columnGap     : 'M',
+        columns       : undefined,
+        customColumns : undefined,
+        customRows    : undefined,
+        hasFullWidth  : false,
+        justify       : 'left',
         role          : undefined,
-        spacing       : 'M',
-        verticalAlign : 'auto',
+        rowGap        : 'M',
+        rows          : undefined,
     };
 
     static displayName = 'Grid';
 
-    static didWarn = {};
-
     render()
     {
-        const deprecatedSpacingOptions =
-            [ 'default', 'h1', 'h2', 'h3', 'h4', 'label' ];
-
         const {
             align,
+            autoCols,
+            autoFlow,
+            autoRows,
             children,
             className,
+            columns,
+            customColumns,
+            customRows,
+            columnGap,
             cssMap = createCssMap( this.context.Grid, this.props ),
-            onClick,
-            onMouseOut,
-            onMouseOver,
-            gutters,
-            hasMinHeight,
-            hasWrap,
-            noWarn,
+            hasFullWidth,
+            justify,
             role,
-            spacing,
-            verticalAlign,
+            rows,
+            rowGap,
         } = this.props;
 
-        if ( !noWarn )
-        {
-            if ( deprecatedSpacingOptions.includes( spacing ) &&
-                !Grid.didWarn[ spacing ] )
-            {
-                console.warn( `Grid spacing option '${spacing}' is depreacted. \
-Please use one of 'S', 'M', 'L' or 'none' instead.` );
-                Grid.didWarn[ spacing ] = true;
-            }
-
-            if ( !Grid.didWarn.hasMinHeight && hasMinHeight !== undefined )
-            {
-                console.warn( 'Grid: \'hasMinHeight\' prop is deprecated. \
-Please use alternative layout.' );
-                this.didWarn.hasMinHeight = true;
-            }
-
-            if ( !Grid.didWarn.children && children !== undefined  )
-            {
-                React.Children.toArray( children ).map( ( child ) =>
-                {
-                    if ( child.type !== Column )
-                    {
-                        console.warn( 'Grid / Row must have Columns as direct \
-children' );
-                        Grid.didWarn.children = true;
-                    }
-                } );
-            }
-        }
+        const layout = {
+            'gridAutoColumns' : autoCols !== undefined ?
```

Comment:
Quotes are not required around the object keys here. Also we probably don't need to explicitly check for `undefined`. I think we can just do: ` gridAutoColumns : autoCols || '1fr'`, for example.